src/graphqlProtocol/subscription.ts

```diff
@@ -0,0 +1,757 @@
+import {
+  GraphQLString,
+  GraphQLID,
+  GraphQLNonNull,
+  GraphQLInputObjectType,
+  GraphQLObjectType,
+  GraphQLInt,
+  GraphQLInputFieldConfigMap,
+  GraphQLFieldConfigMap,
+} from 'graphql';
+import { fromGlobalId } from 'graphql-relay';
+import * as _ from 'lodash';
+
+import { ProtocolGraphQL } from './ProtocolGraphQL';
+import { ProtocolGraphQLConfiguration } from './ProtocolGraphQLConfiguration';
+import { getEntityUniquenessAttributes } from './helper';
+import {
+  getSubscriptionResolver,
+  getSubscriptionPayloadResolver,
+} from './resolver';
+import { isEntity } from '../engine/entity/Entity';
+
+const i18nInputFieldTypesCache = {};
+
```

Comment:
Lots of code duplicated from mutation ; maybe some refactor might be needed by providing generic function usable for subscription and mutation.